stdlib/sys.mli

```diff
@@ -240,110 +249,134 @@ type signal_behavior =
+val set_signal : signal -> signal_behavior -> unit
 (** Same as {!Sys.signal} but return value is ignored. *)
 
-
 (** {2 Signal numbers for the standard POSIX signals.} *)
 
-val sigabrt : int
+val sigabrt : signal
 (** Abnormal termination *)
 
-val sigalrm : int
+val sigalrm : signal
 (** Timeout *)
 
-val sigfpe : int
+val sigfpe : signal
 (** Arithmetic exception *)
 
-val sighup : int
+val sighup : signal
 (** Hangup on controlling terminal *)
 
-val sigill : int
+val sigill : signal
 (** Invalid hardware instruction *)
 
-val sigint : int
+val sigint : signal
 (** Interactive interrupt (ctrl-C) *)
 
-val sigkill : int
+val sigkill : signal
 (** Termination (cannot be ignored) *)
 
-val sigpipe : int
+val sigpipe : signal
 (** Broken pipe *)
 
-val sigquit : int
+val sigquit : signal
 (** Interactive termination *)
 
-val sigsegv : int
+val sigsegv : signal
 (** Invalid memory reference *)
 
-val sigterm : int
+val sigterm : signal
 (** Termination *)
 
-val sigusr1 : int
+val sigusr1 : signal
 (** Application-defined signal 1 *)
 
-val sigusr2 : int
+val sigusr2 : signal
 (** Application-defined signal 2 *)
 
-val sigchld : int
+val sigchld : signal
 (** Child process terminated *)
 
-val sigcont : int
+val sigcont : signal
 (** Continue *)
 
-val sigstop : int
-(** Stop *)
+val sigstop : signal
+(** Stop (cannot be caught or ignored) *)
 
-val sigtstp : int
+val sigtstp : signal
 (** Interactive stop *)
 
-val sigttin : int
+val sigttin : signal
 (** Terminal read from background process *)
 
-val sigttou : int
+val sigttou : signal
 (** Terminal write from background process *)
 
-val sigvtalrm : int
+val sigvtalrm : signal
 (** Timeout in virtual time *)
 
-val sigprof : int
+val sigprof : signal
 (** Profiling interrupt *)
 
-val sigbus : int
+val sigbus : signal
 (** Bus error
     @since 4.03 *)
 
-val sigpoll : int
+val sigpoll : signal
 (** Pollable event
     @since 4.03 *)
 
-val sigsys : int
+val sigsys : signal
 (** Bad argument to routine
     @since 4.03 *)
 
-val sigtrap : int
+val sigtrap : signal
 (** Trace/breakpoint trap
     @since 4.03 *)
 
-val sigurg : int
+val sigurg : signal
 (** Urgent condition on socket
     @since 4.03 *)
 
-val sigxcpu : int
+val sigxcpu : signal
 (** Timeout in cpu time
     @since 4.03 *)
 
-val sigxfsz : int
+val sigxfsz : signal
 (** File size limit exceeded
     @since 4.03 *)
 
+val sigio : signal
+(** I/O is possible on a descriptor
+    @since 5.4 *)
+
+val sigwinch : signal
+(** Window size change
+    @since 5.4 *)
+
+val signal_to_string : signal -> string
+(** [signal_to_string] formats an OCaml [signal] as a C POSIX
+    {{:http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/signal.h.html}
+    constant} or ["SIG(%d)"] for an unrecognised signal number.
+    @since 5.4 *)
+
+val signal_of_int : int -> signal
+(** [signal_of_int n] converts a platform dependent signal number [n] to
+    an OCaml signal number.
+    This is [n] itself if the number is unknown.
```

Comment:
Since POSIX interface generally use `int` for signal numbers on the C side and this is what is used by the conversion functions this is not exactly true on 64-bit platforms. The integers are going to be truncated you should get e.g. `Sys.signal_of_int max_int = Int32.(to_int max_int)`. Not sure it's worth mentioning though.

Reply:
I agree. Using large int values for signal numbers is unusual.